lesson_6/eva_chitul/functions_dataset.py

```diff
@@ -0,0 +1,137 @@
+    ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 b', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+    ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+    ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+    ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
+]
+
+
+# 1 Function that prepares the dataset
+
+
+def prep_dataset(years, dataset):
+    country_dict = {country: [] for country, *_ in dataset}
+
+    length_years = len(years[1])
+
+    for num in range(length_years):
+        for country, elements in dataset:
+            if elements[num] != ': ':
+                raw_number = elements[num].split()
+                country_dict[country].append({'year': years[1][num], 'coverage': int(raw_number[0])})
+            else:
+                country_dict[country].append({'year': years[1][num], 'coverage': 0})
+
+    return country_dict
+
+
+full_dataset = prep_dataset(description, raw_data)
+print('This is the revised dataset: ', full_dataset, '\n')
+
+# 2 Function to retrieve data for each year
+
+
+def year_data(dataset, year):
+    year_dict = {year.strip(): []}
+    # print(year_dict, '\n')
+
+    for country, element in dataset.items():
+        for entry in element:
+            if entry['year'].strip() == year:
+                year_dict[entry['year'].strip()].append((country, entry['coverage']))
+
+    return year_dict
+
+
+data_from_year = year_data(full_dataset, '2011')
+print('The data for the year is: ', data_from_year, '\n')
+
+data_from_year = year_data(full_dataset, '2013')
+print('The data for the year is: ', data_from_year, '\n')
+
+
+# 3 Function to retrieve data for each country
+
+
+def country_data(dataset, country):
+    country_dict = {country: []}
+    # print(country_dict, '\n')
+
+    for countries, elements in dataset.items():
+        if countries == country:
+            for entry in elements:
+                country_dict[country].append((entry['year'].strip(), entry['coverage']))
+
+    return country_dict
+
+
+data_from_country = country_data(full_dataset, 'AL')
+print('The data for the country is: ', data_from_country, '\n')
+
+data_from_country = country_data(full_dataset, 'BG')
+print('The data for the country is: ', data_from_country, '\n')
+
+# 4 Function to perform average from an iterable (year or country data)
+
+
+def average_data(dataset_coverage):
+    number_elements = len(dataset_coverage)
+    if number_elements == 0:
+        return 'No elements in list. Cannot perform average'
+    sum_coverage = 0
+
+    for element in dataset_coverage:
```

Comment:
aici poti incerca `sum()`  https://docs.python.org/3/library/functions.html#sum si o sa scapi de `for`